lib/services/Teleblitz/telbz_firestore.dart

```diff
@@ -5,66 +5,64 @@ import 'package:morea/services/crud.dart';
+  
   Future<bool> eventIDExists(String eventID);
+  //Returns true if a given eventID exists in the firestore
 
-  Future<void> uploadTelbzAkt(String groupnr, Map<String, dynamic> data);
+  Future<void> uploadTelbzAkt(String groupID, Map<String, dynamic> data);
+  /*Uploads the eventID to a specified group. The eventID is stored as a Map. 
+  Key is homeFeed while the eventID is stored as a array as value*/
   Future<void> uploadTelbz(String eventID, Map<String, dynamic> data);
+  //Stores the teleblitz (event content) with the given eventID
 }
 
 class TeleblizFirestore implements BaseTeleblitzFirestore {
   CrudMedthods crud0;
   Map<String, dynamic> _teleblitze = new Map<String,dynamic>();
   Map<String,List<String>> mapHomeFeed = new Map<String,List<String>>();
-  Map<String,Map<String, Map<String, dynamic>>> mapOfGroupEvent = new Map<String,Map<String, Map<String, dynamic>>>();
  
+  Map<String,Map<String, Map<String, dynamic>>> mapOfGroupEvent = new Map<String,Map<String, Map<String, dynamic>>>();
   StreamController<Map<String, List<String>>> _mapHomeFeedController = new BehaviorSubject();
   StreamController<Map<String, Map<String, Map<String,dynamic>>>> _mapofEventsController = new BehaviorSubject();
-  
   Stream<Map<String, List<String>>> get getMapHomeFeed => _mapHomeFeedController.stream;
   Stream<Map<String, Map<String, Map<String,dynamic>>>> get getMapofEvents => _mapofEventsController.stream;
 
   TeleblizFirestore(Firestore firestore, List<String> groupIDs) {
     crud0 = CrudMedthods(firestore);
     _mapHomeFeedController.addStream(this.streamMapHomeFeed(groupIDs));
-    _mapofEventsController.addStream(this.streamMapofGroupEvents(groupIDs));
+    this.streamMapofGroupEvents(groupIDs).first;
   }
 
-
-  Stream<List<String>> streamHomeFeed(String groupID)async*{
+  Stream<Map<String,List<String>>> streamHomeFeed(String groupID)async*{
     Stream<DocumentSnapshot> sDhomeFeed = crud0.streamDocument(pathGroups, groupID); 
-    yield* sDhomeFeed.map((DocumentSnapshot dsHomeFeed){
+    await for(List<String>homeFeed in sDhomeFeed.map((DocumentSnapshot dsHomeFeed){
        return new List<String>.from(dsHomeFeed.data[groupMapHomeFeed]?? new List());
-    }); 
-  }
-  Stream<Map<String,List<String>>> somestream(String groupID)async*{
-    await for(List<String>homeFeed in this.streamHomeFeed(groupID)){
+          })){
         mapHomeFeed[groupID]=homeFeed;
         yield mapHomeFeed;
       }
   }
   Stream<Map<String, List<String>>> streamMapHomeFeed(List<String> groupIDs)async*{
-    List<Stream<Map<String, List<String>>>> streamList = new List();
-    for(String groupID in groupIDs){   
-      streamList.add(somestream(groupID));
-    }
-    yield* StreamGroup.merge(streamList).map((convert){
-      print(convert);
-      return convert;
-    });
+    yield* StreamGroup.merge(groupIDs.map((groupID){
+      return streamHomeFeed(groupID);
+    }));
   }
+
   Stream<Map<String,dynamic>> steramTelebliz(eventID)async*{
```

Comment:
schreibfehler:
```suggestion
  Stream<Map<String,dynamic>> streamTelebliz(eventID)async*{
```
"stream" anstatt "steram"